### PR TITLE
Use plugin parent POM

### PR DESCRIPTION
**Pull request by basil**
Updates the parent POM to the latest version as well as adding a `CODEOWNERS` file and updating the `.gitignore` file to match the latest archetype. See self-review for details.

CC @dwnusbaum

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -1,22 +1,23 @@
...
   <groupId>org.kohsuke</groupId>
   <artifactId>groovy-sandbox</artifactId>
   <version>${revision}${changelist}</version>
+  <url>https://github.com/jenkinsci/${project.artifactId}</url>
 
   <name>Groovy Sandbox</name>
   <description>Executes untrusted Groovy script safely</description>
 
   <properties>
-    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
```
Comment (basil) [inline pom.xml:19]: Part of the parent POM as of jenkinsci/pom#257.

File: pom.xml [hunk 2]
```diff
@@ -51,12 +52,6 @@
           </execution>
         </executions>
       </plugin>
-      <plugin>
-        <artifactId>maven-release-plugin</artifactId>
-        <configuration>
-          <releaseProfiles>release</releaseProfiles>
-        </configuration>
-      </plugin>
```
Comment (basil) [inline pom.xml:59]: The parent POM uses a profile named `jenkins-release` as of jenkinsci/pom#296.

File: pom.xml [hunk 3]
```diff
@@ -94,18 +89,12 @@
 
   <profiles>
     <profile>
-      <id>release</id>
```
Comment (basil) [inline pom.xml:97]: The parent POM uses a profile named `jenkins-release` as of jenkinsci/pom#296.

File: pom.xml [hunk 4]
```diff
@@ -94,18 +89,12 @@
...
         <plugins>
           <plugin>
             <artifactId>maven-source-plugin</artifactId>
             <executions>
-              <execution>
-                <id>attach-sources</id>
-                <goals>
-                  <goal>jar-no-fork</goal>
-                </goals>
-              </execution>
```
Comment (basil) [inline pom.xml:108]: The parent POM's `jenkins-release` profile includes this logic as of jenkinsci/pom#296. However, the logic to attach the test sources is still necessary. I tested this by running `mvn clean verify -Pjenkins-release` and verified that the sources and test sources still got generated.

File: pom.xml [hunk 5]
```diff
@@ -114,17 +103,6 @@
...
-            <artifactId>maven-javadoc-plugin</artifactId>
-            <executions>
-              <execution>
-                <id>attach-javadocs</id>
-                <goals>
-                  <goal>jar</goal>
-                </goals>
-              </execution>
-            </executions>
-          </plugin>
```
Comment (basil) [inline pom.xml:127]: The parent POM's `jenkins-release` profile includes this logic as of jenkinsci/pom#296. I tested this by running `mvn clean verify -Pjenkins-release` and verified that the Javadoc still got generated.

**Comment by basil** [comment]
Per https://github.com/dwnusbaum/groovy-cps/pull/1#issuecomment-1297623554 we actually want to use the plugin parent POM here.

**Review by dwnusbaum — APPROVED** [review]

**Pull request merged**